frontend/src/pages/EditRolesPage/components/MemberTable/index.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect } from 'react';
+import { format, parseISO } from 'date-fns';
+
+import EditRolesButton from '../buttons/EditRolesButton';
+
+import { Loading, RequestErrorSnackbar, Table } from '@SharedComponents';
+import { useRequest } from '@Hooks';
+import { getMultipleInductionClasses } from '@Services/InductionClassService';
+import {
+  MultipleInductionClassResponse,
+  InductionClassResponse,
+  AppUserResponse,
+  MultipleAppUserResponse,
+} from '@Services/api';
+import { inductionClassDateFormat } from '@Constants/dateTimeFormat';
+import { getMultipleUsers } from '@Services/UserService';
+
+// Do Use Effect so table reloads
+
+const appUserResponseToAppUserRow = ({
+  id,
+  firstName,
+  lastName,
+  role,
+  email,
+}: AppUserResponse) => {
+  const userRow = {
+    id,
+    firstName,
+    lastName,
+    name: firstName + ' ' + lastName,
+    email,
+    role,
+  };
+
+  return userRow;
+};
+
+function MemberTable(): JSX.Element {
+  const { data, isLoading, error, isError } = useRequest<
```

Comment:
can we push all the `useRequest` calls up to the page level component, then pass in the responses down to the table as a prop? makes things easier to read

Reply:
In hindsight yeah this is a better change. I remember writing this when I was first learning react. Good catch